pie/storage/__init__.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+from nextcord.ext import commands
+
+from pie.storage.database import Storage
+
+
+def get_str(
+    module: commands.Cog, key: str, guild_id: int = 0, def_val: str = None
+) -> str:
+    value = Storage.get(module.qualified_name, key, guild_id)
+    if not value:
+        return def_val
+
+    return value
+
+
+def get_int(
+    module: commands.Cog, key: str, guild_id: int = 0, def_val: int = None
+) -> int:
+    value = Storage.get(module.qualified_name, key, guild_id)
+    if not value:
+        return def_val
+
+    try:
+        value = int(value)
+    except ValueError as e:
+        if def_val:
+            return def_val
+        else:
+            raise e
+
+    return value
+
+
+def get_float(
+    module: commands.Cog, key: str, guild_id: int = 0, def_val: float = None
+) -> float:
+    value = Storage.get(module.qualified_name, key, guild_id)
+    if not value:
+        return def_val
+
+    try:
+        value = float(value)
+    except ValueError as e:
+        if def_val:
+            return def_val
+        else:
+            raise e
+
+    return value
+
+
+def get_bool(
+    module: commands.Cog, key: str, guild_id: int = 0, def_val: bool = None
+) -> bool:
+    value = Storage.get(module.qualified_name, key, guild_id)
+    if not value:
+        return def_val
+
+    if value.lower() not in ["true", "false"]:
+        if def_val:
+            return def_val
+
+        raise ValueError(f"Value {value} for key {key} is not bool!")
+
+    value = value.lower() == "true"
+
+    return value
+
+
+def set(module: commands.Cog, key: str, value: object, guild_id: int = 0) -> bool:
+    return Storage.set(module.qualified_name, key, value, guild_id) is not None
+
+
+def default(module: commands.Cog, key: str, value: object, guild_id: int = 0) -> bool:
+    return (
+        Storage.set(module.qualified_name, key, value, guild_id, ignore_existing=True)
+        is not None
+    )
```

Comment:
What's this function's task?

Reply:
`set_if_missing`? I'm confused by `default`, it does not properly describe what it's doing.

Reply:
OK, that could do.